src/migtd/src/bin/migtd/main.rs

```diff
@@ -142,10 +142,29 @@ fn handle_pre_mig() {
 
             if let Some(request) = new_request {
                 async_runtime::add_task(async move {
-                    let status = exchange_msk(&request)
-                        .await
-                        .map(|_| MigrationResult::Success)
-                        .unwrap_or_else(|e| e);
+ 
+                    // Determine the status based on enabled features
+                    let status = {
+                        #[cfg(feature = "test_force_success_status")]
+                        {
+                            // Execute exchange_msk but ignore the result
+                            let _result = exchange_msk(&request).await;
+                            MigrationResult::Success
+                        }
```

Comment:
Why not returning the exchange_msk() result as the status, as the original code?